src/commands/serve.ts

```diff
@@ -34,15 +35,20 @@ export class ServeCommand {
     if (this.globalOptions.env) {
       pod.setEnvironment(this.globalOptions.env);
     }
-    const watcher = new Watcher(pod);
-    const app = createApp(pod);
-    app.listen(port, () => {
+    // Create server.
+    const server = new Server(pod, {
+      port: port,
+    });
+    server.start();
+    server?.httpServer?.once('listening', () => {
```

Comment:
I don't think you should expose `httpServer` as a public member. That'll make it harder to maintain long-term. Either move this callback to the `start()` method or provide a callback handler like `server.on('start', () => console.log('server started'))`.

I would also suggest moving the console logging for restarts here as well so that it's all in the same spot.

Reply:
Made Server an EventEmitter and emitted events so they can be handled and logged from within the command